fhevm/operators_arithmetic.go

```diff
@@ -3,11 +3,193 @@ package fhevm
 import (
 	"encoding/hex"
 	"errors"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"go.opentelemetry.io/otel/trace"
 )
 
+func sgxAddRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	input = input[:minInt(65, len(input))]
+
+	logger := environment.GetLogger()
+
+	lhs, rhs, err := get2VerifiedOperands(environment, input)
+	otelDescribeOperands(runSpan, encryptedOperand(*lhs), encryptedOperand(*rhs))
+	if err != nil {
+		logger.Error("fheAdd inputs not verified", "err", err, "input", hex.EncodeToString(input))
+		return nil, err
+	}
+	if lhs.fheUintType() != rhs.fheUintType() {
+		msg := "fheAdd operand type mismatch"
+		logger.Error(msg, "lhs", lhs.fheUintType(), "rhs", rhs.fheUintType())
+		return nil, errors.New(msg)
+	}
+
+	// If we are doing gas estimation, skip execution and insert a random ciphertext as a result.
+	if !environment.IsCommitting() && !environment.IsEthCall() {
+		return importRandomCiphertext(environment, lhs.fheUintType()), nil
+	}
+
+	lb, err := decryptRun(environment, caller, addr, input[0:32], readOnly, runSpan)
```

Comment:
We shouldn't use decryptRun (use FHE key). We should create a new `sgxDecryptRun`. Under the hood, it can use a mocked key.

For example, we can create a `sgx/mock.go` file, with this:

```golang
const MockPrivateKey = []byte(...) // some randome bytes

func SgxDecryptRun(env, caller, input) {
  // pseudo-code
  return MockPrivateKey.Decrypt(input)
}